src/main/java/duke/command/FindCommand.java

```diff
@@ -0,0 +1,86 @@
+package duke.command;
+
+import java.util.ArrayList;
+
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+import duke.exception.DukeBadInputException;
+import duke.task.Task;
+
+/**
+ * Find a task by searching for a keyword.
+ */
+public class FindCommand extends Command {
+
+    /**
+     * Stores the search query
+     */
+    private String query;
+
+    /**
+     * Constructor of the todo command
+     *
+     * @param query - the desc of the command
+     */
+    public FindCommand(String query) {
+        this.query = query;
+    }
+
+    /**
+     * Method to encapsulate the execution logic of the command
+     *
+     * @param taskList - the task list instance  of the current duke
+     * @param ui       - the ui instance of DUKE
+     * @param storage  - the storage instance to allow the command to write to the storage
+     * @return the reply of Quack
+     * @throws DukeBadInputException - if the input cannot be used
+     */
+    @Override
+    public String execute(TaskList taskList, Ui ui, Storage storage) throws DukeBadInputException {
+        ArrayList<Task> matches = new ArrayList<>();
+        String find = this.query.toUpperCase();
+
+        // filter out relevant
+        for (Task t : taskList.getAllTask()) {
+            if (t.getTask().toUpperCase().contains(find)) {
+                matches.add(t);
+            }
+        }
+
+        // Handles no matches
```

Comment:
Nice use of additional comments to help further explain how the function works! 👍